nselib/url.lua

```diff
@@ -177,8 +178,22 @@ function parse(url, default)
     parsed.params = p
     return ""
   end)
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
+    parsed.is_folder = true
+  else
+    parsed.is_folder = false
+    local split_str = stdnse.strsplit('%.', parsed.path)
+    if(split_str and #split_str > 1) then
+      parsed.extension = stdnse.strsplit("%;", split_str[#split_str])[1]
+    end
```

Comment:
@nnposter , your proposed regex `parsed.path:match("%.([^/;]+)%f[;\0][^/]*$")` doesn't work for some urls like `http://php.net/manual/en/function.pathinfo.php`. It stores extension as `pathinfo.php` which is not.  So, here I propose a new regex.

```
parsed.extension = parsed.path:match("/[^/]+%.([^%.]+)$")
```
But this regex doesn't work on some urls like `https://svn.nmap.org/nmap-releases/nmap-7.60`. It stores extension as `60` which is not. To handle these kind of false positives, I wrote a new regex.
```
parsed.extension = parsed.path:match("/[^/]+%.([^%.0-9]+)$")
```
If we are sure that the extensions will be only ASCII characters then we can use the above regex.
Your views?